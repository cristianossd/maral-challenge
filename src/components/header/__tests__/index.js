import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Header from '../';

describe('<Header />', () => {
  it('render correctly', () => {
    const wrapper = shallow(<Header />);

    expect(wrapper.find('.Header').length).toBe(1);
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    const links = wrapper.find('Link');
    expect(links.length).toBe(5);
    expect(links.at(0).prop('to')).toEqual('/resources');
    expect(links.at(1).prop('to')).toEqual('/teams/generate');
    expect(links.at(2).prop('to')).toEqual('/events/score');
    expect(links.at(3).prop('to')).toEqual('/events/ranking');
    expect(links.at(4).prop('to')).toEqual('/teams/score');
  });
});
