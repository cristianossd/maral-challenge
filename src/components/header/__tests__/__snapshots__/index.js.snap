// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render correctly 1`] = `
<div
  className="Header"
>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <Link
      className="nav-link"
      replace={false}
      to="/resources"
    >
      Maral Challenge
    </Link>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav"
      >
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/teams/generate"
          >
            Equipes
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/events/score"
          >
            Resultados
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/events/ranking"
          >
            Scores por Evento
          </Link>
        </li>
        <li
          className="nav-item"
        >
          <Link
            className="nav-link"
            replace={false}
            to="/teams/score"
          >
            Ranking Geral
          </Link>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
